### ISSUE TITLE
Doxygen 1.8.17 false warnings and no generate output for documented const functions

### ISSUE THREAD

**Issue by neg** [doxygen/doxygen#7585]
Starting with 1.8.17 there are warnings about missing documentation for const functions that in fact are documented. The documentation is also missing in the generate output. The same input source code generates the expected output and no warnings on 1.8.16.

The issue can be demonstrated with an example.

> cat example.cpp
```
class Example
{
public:
	int foo();
	int foo() const;
};

/**
 * \brief Foo
 */
int Example::foo()
{
	return 0;
}

/**
 * \brief Const foo
 */
int Example::foo() const
{
	return 0;
}
```

Generate doxygen:
> doxygen -g config
> doxygen config

Warning that the documentation is missing.

> example.cpp:5: warning: Member foo() const (function) of class Example is not documented.

I bisected the problem to commit c38a6fe88ceb3658 ("Changed ArgumentList to be an STL container").

**Comment by doxygen** [comment]
@neg Please verify that the referenced commit fixes the problem. Please keep this report open. It will be closed when a new release becomes available.

**Comment by neg** [comment]
I tired latest master 3b392af822d81ae5 ("Merge pull request #7601 from albert-github/feature/bug_rtf_list") which contains 8075ba0bbe45e169 ("issue #7585: Doxygen 1.8.17 false warnings and no generate output for documented const functions") and it solves my problem.

Thanks for your work!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.18.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
